src/share/poudriere/common.sh

```diff
@@ -3869,14 +3869,15 @@ save_wrkdir() {
 	tgz) COMPRESSKEY="z" ;;
 	tbz) COMPRESSKEY="j" ;;
 	txz) COMPRESSKEY="J" ;;
+	tzst) COMPRESSKEY="-zstd" ;;
 	esac
 	unlink ${tarname}
 
 	port_var_fetch_originspec "${originspec}" \
 	    WRKDIR wrkdir || \
 	    err 1 "Failed to lookup WRKDIR for ${originspec}"
 
-	tar -s ",${mnt}${wrkdir%/*},," -c${COMPRESSKEY}f "${tarname}" \
+	tar -s ",${mnt}${wrkdir%/*},," -cf "${tarname}" -${COMPRESSKEY} \
```

Comment:
`tar) COMPRESSKEY=""` means this line turns into `-` at the end. Does that work?
Otherwise this would add the `-` if not blank:
```
${COMPRESSKEY:+-${COMPRESSKEY}}
```